extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -328,6 +373,129 @@ class LifecycleBucketProcessor {
         });
     }
 
+    /**
+     * Connects to MongoDB using and retreives
+     * the location status collection
+     * @param {function} cb callback
+     * @returns {undefined} undefined
+     */
+    _setupMongoClient(cb) {
+        const mongoUrl = constructConnectionString(this._mongoConfig);
+        MongoClient.connect(mongoUrl, {
+                replicaSet: this._mongoConfig.replicaSet,
+                useNewUrlParser: true,
+                useUnifiedTopology: true,
+        }, (err, client) => {
+            if (err) {
+                this._log.error('Could not connect to MongoDB', {
+                    method: 'LifecycleBucketProcessor._setupMongoClient',
+                    error: err.message,
+                });
+                return cb(err);
+            }
+            // connect to metadata DB
+            this._mongoClient = client.db(this._mongoConfig.database, {
+                ignoreUndefined: true,
+            });
+            // creating a collection in case it doesn't exist, this is done as a safety measure
+            // but normally it will be created by backbeat API that gets deployed before the
+            // lifecycle bucket processor.
+            return this._mongoClient.createCollection(locationStatusCollection, err => {
```

Comment:
> locationStatusCollection

don't we need to add "filters" in other parts of Backbeat, to ensure this collection is not unexpectedly processed by replication/ingestion/... ?

Reply:
I think there's no need for that, in change stream implementation we only listen to buckets defined on the metastore, and in the oplog implementation, this would still be ignored as we only replicate proper objects that have a replication status.